tslearn/clustering.py

```diff
@@ -30,6 +33,99 @@
 # Mathieu Blondel, under BSD 3 clause license
 
 
+def _k_init_metric(X, n_clusters, cdist_metric, random_state,
+                   n_local_trials=None):
+    """Init n_clusters seeds according to k-means++ with a custom distance
+    metric.
+
+    Parameters
+    ----------
+    X : array or sparse matrix, shape (n_samples, n_features)
```

Comment:
The shape of a dataset is different in tslearn and I don't think that sparse matrices are supported